### PR TITLE
Moving Github Source to contrib folder

### LINKED ISSUE #423: Move GitHubSource to contrib
## Expected Behavior

Sources are organized and installed consistently. GitHubSource, like other sources, lives in `contrib/github` and is installed by applying `github.yaml`.

## Actual Behavior

All sources are in `contrib/<sourcename>` except GitHubSource which is in the root dir. All sources are installed with `<sourcename>.yaml` except GitHubSource which is installed with the inaccurately-named `eventing-sources.yaml`.

/kind chore

### PR DESCRIPTION
**Pull request by nachocano**
Fixes #423

## Proposed Changes

  * Moving github source to contrib folder.

**Release Note**

```release-note
GitHubSource moved to `contrib` folder. You need to `kubectl apply contrib/github/config` to get the new controller running.
```

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -57,15 +57,9 @@ follows.
...
-Once you've [setup your development environment](#getting-started), install
-sources _Github Source_ and _AWS SQS Source_ with:
-
-```shell
-ko apply -f config/
-```
-
-or install a single source _Camel Source_, _Gcppubsub Source_, _Kafka Source_
-with
+Once you've [setup your development environment](#getting-started), install any of the sources
```
Comment (mattmoor-sockpuppet) [inline DEVELOPMENT.md]:
```suggestion
Once you've [setup your development environment](#getting-started), install any of the sources
```

Comment (mattmoor-sockpuppet) [inline DEVELOPMENT.md]:
```suggestion
Once you've [setup your development environment](#getting-started), install any of the sources
```

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -131,7 +125,7 @@ Running tests as you make changes to the code-base is pretty simple. See
 You can delete `Knative Sources` with:
```
Comment (grantr) [inline DEVELOPMENT.md]: nit: This should probably say `You can delete a source with:`

**Comment by grantr** [comment]
As part of the upgrade instructions, we'll probably also need to tell the user to delete the old `eventing-sources/controller-manager` StatefulSet, right? Otherwise the controllers will fight.

**Comment by nachocano** [comment]
> As part of the upgrade instructions, we'll probably also need to tell the user to delete the old `eventing-sources/controller-manager` StatefulSet, right? Otherwise the controllers will fight.

Yes, thanks! Added that in the release notes.

**Pull request merged**